### DISCUSSION TITLE
Uppercase for "cmake.buildDirectory": "${workspaceFolder}/BUILD/${variant:buildType}/BUILD"

### DISCUSSION THREAD

**Discussion by JohannesRoll** [microsoft/vscode-cmake-tools#2841] [Q&A] [answered]
Is there a way to make ${variant:buildType} all uppercase because I work in existing projects where these folders (in all uppercase) already exist.

**Comment by bobbrow** [comment, marked as answer]
Not currently.  We'd have to support some kind of new syntax to make that work.  Is your filesystem case-sensitive?

**Reply by JohannesRoll** [reply]
Yes, the filesystem is case-sensitive, a solution where you could just set the value of ${variant:buildType} in a file would be really nice.

**Reply by bobbrow** [reply]
It looks like there may be another way.  If you use all uppercase labels for your options, I think you should get the behavior you want.  For example, this worked for me:

```
    buildType:
      default: DEBUG
      choices:
        DEBUG:
          short: DEBUG
          long: Emit debug information
          buildType: Debug
        RELEASE:
          short: RELEASE
          long: Optimize generated code
          buildType: Release
```

Substituting `${variant:buildType}` gave me 'DEBUG' or 'RELEASE' and the correct 'Debug' or 'Release' build type was sent to CMake.

**Reply by JohannesRoll** [reply]
Where can I set these options?

**Reply by bobbrow** [reply]
Sorry, I assumed you had been creating your own variants because you were using the ${variant:\<foo\>} syntax.  The documentation is here: https://github.com/microsoft/vscode-cmake-tools/blob/main/docs/variants.md

Basically, create a `cmake-variants.yaml` file in your `.vscode` folder.

**Reply by JohannesRoll** [reply]
Thank you very much, with  your second to last tip I just found it also worked for me by replacing  debug with DEBUG etc. in Line 1363, 1366, 1371, 1376, 1381 in the package.json file in .vscode/extensions/ms-vscode.cmake-tools-1.9.2

**Reply by bobbrow** [reply]
I suppose you could do that too, but we don't recommend changing the internal code directly because you'll lose those changes the next time you update the extension.  v1.9.2 is a pretty old version of the extension, BTW.  We're on 1.12 and close to releasing 1.13.
